initramfs/src/registry.rs

```diff
@@ -1,35 +1,138 @@
 use std::io::Cursor;
 
 use anyhow::{anyhow, Result};
-use log::{debug, info};
+use log::{debug, info, trace};
 use serde::Deserialize;
 
 use crate::{
     httpclient::run_get_request,
     image::{Image, Layer, LayerMetadata},
 };
 
+#[derive(Deserialize, Debug)]
+struct ManifestListV2Response {
+    manifests: Vec<ManifestV2ItemResponse>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct ManifestV2Response {
+    schema_version: u32,
+    media_type: String,
+    config: Config,
+    layers: Vec<LayerInfo>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct OCIv1ListResponse {
+    media_type: String,
+    schema_version: u32,
+    manifests: Vec<LayerInfo>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+pub struct Config {
+    media_type: String,
+    size: u64,
+    digest: String,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct LayerInfo {
+    digest: String,
+    media_type: String,
+    platform: Option<Platform>,
+    size: u64,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct ManifestV2ItemResponse {
+    digest: String,
+    media_type: String,
+    platform: Option<Platform>,
+    size: u64,
+}
+
+#[derive(Deserialize, Debug)]
+#[allow(dead_code)]
```

Comment:
there is a lot of #[allow(dead_code)] in this file, is this normal?

Reply:
These structs are constructed from responses from an API. I wanted to be exhaustive in the fields that we get back from the API, even if we currently use these fields. However, clippy sees this as a mistake, hence the `#[allow(dead_code)]` above these structs.

Wether or not I should have done it this way is debatable. Do you think this is a mistake ?

Reply:
Ok, knowing that it is constructed in order to know more about what is available on the this API I understand better. 
IMO: if you think that we will need to modify this part of the code in a near future and that it will require to use those fields, it's better to keep it this way. Otherwise I think it's better to keep the code to the minimal quantity possible in order to not be confusing.

Reply:
Makes sense to me. I don't think we will be using these fields, I'll remove them.